JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
 package ru.javafiddle.core.ejb;
 
-import javax.ejb.EJB;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
 import javax.persistence.NoResultException;
 import javax.persistence.PersistenceContext;
 import javax.ejb.Stateless;
 import javax.persistence.EntityManager;
+import javax.ws.rs.client.Entity;
 
-import ru.javafiddle.jpa.entity.Access;
-import ru.javafiddle.jpa.entity.Group;
-import ru.javafiddle.jpa.entity.Project;
-import ru.javafiddle.jpa.entity.Hash;
-import ru.javafiddle.jpa.entity.User;
-import ru.javafiddle.jpa.entity.UserGroup;
+import ru.javafiddle.jpa.entity.*;
 
 import java.io.UnsupportedEncodingException;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 
 @Stateless
 public class ProjectBean {
 
+    private static final Logger logger =
+            Logger.getLogger(ProjectBean.class.getName());
+
     private static final String DEFAULT_GROUP_NAME = "default";
 
-    @PersistenceContext
+    @PersistenceContext(name = "JFPersistenceUnit")
     EntityManager em;
 
-    @EJB
-    UserBean userBean;
+    public ProjectBean() {}
 
-    public ProjectBean() {
-    }
+    //services : WITHOUT GROUPNAME WE CANNOT CREATE SERVICES AS WE NEED TO UPDATE PROJECT LIST IN GROUP
+    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
+    public Project createProject(int groupId, String hash, String projectName) throws UnsupportedEncodingException, NoSuchAlgorithmException {//groupName??
```

Comment:
As I've already commented, from the beginning we wanted to separate at maximum web services and ebb beans, so creating entity instances to pass them as parameters in web services is not our goal.
23 марта 2016 г., в 2:30, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java:
> 
> >  
> > -    public ProjectBean() {
> > -    }
> > +    //services : WITHOUT GROUPNAME WE CANNOT CREATE SERVICES AS WE NEED TO UPDATE PROJECT LIST IN GROUP
> > +    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
> > +    public Project createProject(int groupId, String hash, String projectName) throws UnsupportedEncodingException, NoSuchAlgorithmException {//groupName??
> Please consider passing a Project entity as the input
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>